aya/src/programs/lirc_mode2.rs

```diff
@@ -101,12 +101,12 @@ impl LircMode2 {
                 io_error,
             })?;
 
-            prog_fds.push(fd as RawFd);
+            prog_fds.push(fd);
         }
 
         Ok(prog_fds
             .into_iter()
-            .map(|prog_fd| LircLink::new(prog_fd, target_fd.as_raw_fd()))
+            .map(|prog_fd| LircLink::new(prog_fd.into_raw_fd(), target_fd.as_raw_fd()))
```

Comment:
@tamird 

I know this is already merged but the `OwnedFd` is dropped here since there are no other usages of `OwnedFd` and thus the file descriptor is closed and yet we are creating a `LircLink` with this closed file descriptor. `LircLink` has a method `info` that uses this file descriptor to get its program information but that file descriptor is now closed. I don't really use lirc probes so I am not how how to test/prove that but I figured it's worth calling out that this PR most likely broke getting the program information from a queried list of lirc programs.

Reply:
Derp nvm, I thought we were calling for `as_raw_fd`. Ignore